### PR TITLE
Add scheduler crd

### PR DESCRIPTION
**Pull request by CoderKevinZhang**
**What type of PR is this?**
> /kind design

**What this PR does / why we need it**:
This PR is for creating a folder called globalscheduler to store CRDs and controllers

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by kimeunju108 — COMMENTED** [review]
I am just curious whether importpath is 
importpath = "k8s.io/globalscheduler/controllers",  or
importpath = "k8s.io/kubernetes/globalscheduler/controllers",

**Review by kimeunju108 — COMMENTED** [review]
Does it need a Status of scheduler as follow ?
Status string      `json:"status"`

type Scheduler struct {
	metav1.TypeMeta   `json:",inline"`
	metav1.ObjectMeta `json:"metadata,omitempty"`

	Spec SchedulerSpec `json:"spec"`
}

**Comment by CoderKevinZhang** [comment]
> I am just curious whether importpath is
> importpath = "k8s.io/globalscheduler/controllers", or
> importpath = "k8s.io/kubernetes/globalscheduler/controllers",

File: globalscheduler/pkg/apis/scheduler/v1/types.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type Scheduler struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
```
Comment (kimeunju108) [inline globalscheduler/pkg/apis/scheduler/v1/types.go:28]: Does it need a Status of scheduler as follow ?
Status string      `json:"status"`

type Scheduler struct {
	metav1.TypeMeta   `json:",inline"`
	metav1.ObjectMeta `json:"metadata,omitempty"`

	Spec SchedulerSpec `json:"spec"`
}

Comment (CoderKevinZhang) [reply globalscheduler/pkg/apis/scheduler/v1/types.go:28]: I think we need status for the scheduler. I will add it. Thx

**Review by kimeunju108 — APPROVED** [review]
LGTM

File: globalscheduler/pkg/apis/scheduler/register.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scheduler
+
+const (
+	GroupName = "globalscheduler.com"
```
[the remark this answers is not in the document]
Comment (CoderKevinZhang) [reply globalscheduler/pkg/apis/scheduler/register.go:20]: This is the decision that Cathy made. Maybe we can change it in the future? What do you think?

File: globalscheduler/controllers/scheduler_controller.go [hunk 3]
```diff
@@ -0,0 +1,17 @@
+/*
+Copyright 2020 The Kubernetes Authors.
```
[the remark this answers is not in the document]
Comment (CoderKevinZhang) [reply globalscheduler/controllers/scheduler_controller.go]: This is the outdated file, the latest one is Copyright 2020 Authors of Arktos. I believe this is auto changed by running `make update`.

File: globalscheduler/pkg/apis/scheduler/v1/types.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+	Spec SchedulerSpec `json:"spec"`
+}
+
+type SchedulerSpec struct {
+	// Name is the name of the scheduler
+	Name string `json:"name"`
+
+	// Cluster is an array that stores the name of clusters
+	Cluster []string `json:"cluster"`
```
[the remark this answers is not in the document]
Comment (CoderKevinZhang) [reply globalscheduler/pkg/apis/scheduler/v1/types.go:37]: This is a good point. Do you think a scheduler can have different types of clusters or all the same types?

[the remark this answers is not in the document]
Comment (jshaofuturewei) [reply globalscheduler/pkg/apis/scheduler/v1/types.go:37]: We are using openstack clusters now. Is that possible that we use kubernetes or arktos clusters in the future?

[the remark this answers is not in the document]
Comment (CoderKevinZhang) [reply globalscheduler/pkg/apis/scheduler/v1/types.go:37]: I think it is possible in the future that global scheduler might need to be compatible with kubernetes or arktos clusters. However, in our first, we can assume that only openstack clusters are supported. What do you think?

**Review by jshaofuturewei — APPROVED** [review]
LGTM

**Pull request merged**